### DISCUSSION TITLE
receive websend from rule on script device

### DISCUSSION THREAD

**Discussion by Kugeleis** [arendst/Tasmota#12794] [Q&A] [answered]
I try to websend from a rule device like so
`WebSend [192.168.2.5] Var8 123`
with having defined 
```
>D
Var8=0
```
on the receiving script device.
but
```
>E
print %Var8%
```
returns 0 only. What's wrong?

**Comment by sfromis** [comment, marked as answer]
I don't know details of how scripting really works, but I'd try without having that define of a variable, as I suspect this may be overriding the Tasmota variable of the same name.

**Reply by Kugeleis** [reply]
This doesn't work. Then it's just undefined.
I guess @arendst would know details about scripting :~)

**Reply by sfromis** [reply]
No

**Reply by gemu2015** [reply]
WebSend [192.168.2.5] script >Var8=123

can you read the docs please:

Console Commands
script <n> : 0 = switch script off; 1 = switch script on
script ><cmdline> execute
- Can be used to set variables, e.g., script >mintmp=15
- Multiple statements can be specified by separating each with a semicolon, e.g. script
\>mintmp=15;maxtemp=40
script?<var> queries a script variable var

**Reply by Kugeleis** [reply]
Thanks @gemu2015 , this actually works. I read the docs but this edge case isn't obvious (and nobody in the forums knew). Maybe it's worth to be mentioned in thedocts that on the rule device the script syntax is needed in the websend payload.
